### ISSUE TITLE
Wrong energy for benzene

### ISSUE THREAD

**Issue by susilehtola** [pyscf/pyscf#1018]
Running the calculation below
```
import pyscf

neutral = pyscf.M(
    atom = '''
C          0.00000        0.00000        1.39495
C          0.00000        1.20806        0.69748
C          0.00000        1.20806       -0.69748
C          0.00000        0.00000       -1.39495
C          0.00000       -1.20806       -0.69748
C          0.00000       -1.20806        0.69748
H          0.00000        0.00000        2.47753
H          0.00000        2.14560        1.23877
H          0.00000        2.14560       -1.23877
H          0.00000        0.00000       -2.47753
H          0.00000       -2.14560       -1.23877
H          0.00000       -2.14560        1.23877
''',
    basis = 'aug-pc-0',
    symmetry = True,
    verbose = 4,
    charge = 0,
    spin = 0
)

mf_neutral = neutral.KS().apply(pyscf.scf.addons.remove_linear_dep_)
mf_neutral.grids.prune = None
mf_neutral.grids.atom_grid = (100, 590)
mf_neutral.xc = 'hyb_gga_xc_bhandh'
mf_neutral.kernel()

anion = pyscf.M(
    atom = '''
C          0.00000        0.00000        1.39495
C          0.00000        1.20806        0.69748
C          0.00000        1.20806       -0.69748
C          0.00000        0.00000       -1.39495
C          0.00000       -1.20806       -0.69748
C          0.00000       -1.20806        0.69748
H          0.00000        0.00000        2.47753
H          0.00000        2.14560        1.23877
H          0.00000        2.14560       -1.23877
H          0.00000        0.00000       -2.47753
H          0.00000       -2.14560       -1.23877
H          0.00000       -2.14560        1.23877
''',
    basis = 'aug-pc-0',
    symmetry = True,
    verbose = 4,
    charge = -1,
    spin = 1
)

mf_anion = anion.KS().apply(pyscf.scf.addons.remove_linear_dep_)
mf_anion.grids.prune = None
mf_anion.grids.atom_grid = (100, 590)
mf_anion.xc = 'hyb_gga_xc_bhandh'
mf_anion.kernel()
```
gives me converged SCF energies -229.871832298617 and -229.835867695121 for the neutral molecule and the anion with PySCF 1.7.6.

Running the calculation with a snapshot of PySCF master branch gave poor convergence for the neutral and energies 10x and 20x too low for the neutral molecule and anion, respectively.... I am trying to recompile and see if that solves the issue

**Comment by susilehtola** [comment]
I recompiled the current master branch, and the aug-pc-0 calculations ran fine.

However, already in aug-pc-1 the calculation goes haywire:
```
init E= -231.101033245861
HOMO (B3g) = -7.7395871345262  LUMO (B3u) = -5.74268052711217
cycle= 1 E= -217163901.058065  delta_E= -2.17e+08  |g|= 2.46e+07  |ddm|= 1.57e+06
HOMO (B2u) = -852700.354299115  LUMO (B1u) = -852697.097562663
cycle= 2 E= -574631980.624751  delta_E= -3.57e+08  |g|= 1.84e+07  |ddm|= 1.37e+05
HOMO (B1u) = -427697.404388843  LUMO (B3g) = -417271.030480679
cycle= 3 E= -645385319.509049  delta_E= -7.08e+07  |g|= 1.93e+07  |ddm|= 6.91e+04
HOMO (B1u) = -256502.426523699  LUMO (B1u) = -173265.071839651
cycle= 4 E= -662487756.619415  delta_E= -1.71e+07  |g|= 2.08e+07  |ddm|= 5.86e+04
HOMO (B1u) = -239952.244014946  LUMO (B1u) = -174924.866586751
```

**Comment by susilehtola** [comment]
This is quite odd since the usual suspect for non-variational results would be an ill-conditioned basis, which is not the case here.

**Comment by susilehtola** [comment]
pc-2 also leads to wildly wrong results
```
init E= -231.629453403121
HOMO (B3u) = -0.43027997750106  LUMO (B3g) = -0.382114431715916
cycle= 1 E= -69676.0076400298  delta_E= -6.94e+04  |g|= 2.6e+04  |ddm|= 1.02e+04
HOMO (Ag) = -6555.91778160011  LUMO (B3g) = -6555.81467048198
cycle= 2 E= -384157.747689078  delta_E= -3.14e+05  |g|= 6.86e+04  |ddm|= 1.65e+04
HOMO (Ag) = -5824.34518008497  LUMO (B3g) = -5790.00400330834
```
and the calculations also fail at the Hartree-Fock level of theory

**Comment by susilehtola** [comment]
Turning off symmetry has no effect

**Comment by susilehtola** [comment]
Reduced test case
```
import pyscf

mol = pyscf.M(
    atom = '''
C          0.00000        0.00000        1.39495
C          0.00000        1.20806        0.69748
C          0.00000        1.20806       -0.69748
C          0.00000        0.00000       -1.39495
C          0.00000       -1.20806       -0.69748
C          0.00000       -1.20806        0.69748
H          0.00000        0.00000        2.47753
H          0.00000        2.14560        1.23877
H          0.00000        2.14560       -1.23877
H          0.00000        0.00000       -2.47753
H          0.00000       -2.14560       -1.23877
H          0.00000       -2.14560        1.23877
''',
    basis = 'pc-2',
    symmetry = True,
    verbose = 4
)

mf = mol.HF()
mf.kernel()
```
result
```
init E= -232.393799745241
HOMO (B3u) = -0.430682215804206  LUMO (B3g) = -0.35307779954462
cycle= 1 E= -78012.7949510677  delta_E= -7.78e+04  |g|= 2.5e+04  |ddm|= 1.02e+04
HOMO (Ag) = -6773.99600039286  LUMO (B3g) = -6773.89456486638
cycle= 2 E= -423067.480828103  delta_E= -3.45e+05  |g|= 6.63e+04  |ddm|= 1.7e+04
HOMO (Ag) = -6183.13637331776  LUMO (B3g) = -6135.99357445351
cycle= 3 E= -486677.0033498  delta_E= -6.36e+04  |g|= 4.7e+04  |ddm|= 1.74e+03
HOMO (Ag) = -6290.19474179859  LUMO (B2g) = -6242.05499997989
cycle= 4 E= -527709.459517495  delta_E= -4.1e+04  |g|= 1.31e+04  |ddm|= 1.58e+03
HOMO (B1u) = -7098.19399704283  LUMO (Ag) = -7023.76484545713
cycle= 5 E= -531649.380799805  delta_E= -3.94e+03  |g|= 3.41e+03  |ddm|=  940
HOMO (Ag) = -7354.02765485832  LUMO (B1u) = -7240.24478604687
```

**Comment by susilehtola** [comment]
This is with qcint; libcint gives the correct result

**Comment by Majokux** [comment]
Possibly related to #953 ?

See eg. comment by @btglau in https://github.com/pyscf/pyscf/issues/953#issuecomment-849057359_

**Comment by sunqm** [comment]
It's a bug in the integral library libcint 4.1.* - 4.4.2 due to a divide-by-0 bug. It is fixed in the new release 4.4.3

**Comment by susilehtola** [comment]
Confirmed, issue is fixed! Thanks!

**Comment by susilehtola** [comment]
Issue still prevails in PySCF 1.7.6 and qcint 4.4.3
```
init E= -232.393799745242
HOMO (B3u) = -0.430682215804213  LUMO (B3g) = -0.35307779954455
cycle= 1 E= -78012.7949510526  delta_E= -7.78e+04  |g|= 2.5e+04  |ddm|= 1.02e+04
HOMO (Ag) = -6773.99600039274  LUMO (B3g) = -6773.89456486676
cycle= 2 E= -423067.480828  delta_E= -3.45e+05  |g|= 6.63e+04  |ddm|= 1.7e+04
HOMO (Ag) = -6183.13637331704  LUMO (B3g) = -6135.99357445377
```

**Comment by susilehtola** [comment]
https://github.com/sunqm/libcint/commit/914caf2278bda61d5fbcecadbe1bccf5aa86ca9f

This looks like something that could cause such results. Checking

**Comment by susilehtola** [comment]
No, there is still definitely an issue with qcint 4.4.4 as well:
```
init E= -232.393799745242
HOMO (B3u) = -0.430682215804207  LUMO (B3g) = -0.35307779954462
cycle= 1 E= -78012.7949510903  delta_E= -7.78e+04  |g|= 2.5e+04  |ddm|= 1.02e+04
HOMO (Ag) = -6773.99600039593  LUMO (B3g) = -6773.89456486988
cycle= 2 E= -423067.480828085  delta_E= -3.45e+05  |g|= 6.63e+04  |ddm|= 1.7e+04
HOMO (Ag) = -6183.13637331985  LUMO (B3g) = -6135.99357445604
cycle= 3 E= -486677.003349847  delta_E= -6.36e+04  |g|= 4.7e+04  |ddm|= 1.74e+03
HOMO (Ag) = -6290.19474179985  LUMO (B2g) = -6242.05499998264
```

**Comment by sunqm** [comment]
> No, there is still definitely an issue with qcint 4.4.4 as well:
> 
> ```
> init E= -232.393799745242
> HOMO (B3u) = -0.430682215804207  LUMO (B3g) = -0.35307779954462
> cycle= 1 E= -78012.7949510903  delta_E= -7.78e+04  |g|= 2.5e+04  |ddm|= 1.02e+04
> HOMO (Ag) = -6773.99600039593  LUMO (B3g) = -6773.89456486988
> cycle= 2 E= -423067.480828085  delta_E= -3.45e+05  |g|= 6.63e+04  |ddm|= 1.7e+04
> HOMO (Ag) = -6183.13637331985  LUMO (B3g) = -6135.99357445604
> cycle= 3 E= -486677.003349847  delta_E= -6.36e+04  |g|= 4.7e+04  |ddm|= 1.74e+03
> HOMO (Ag) = -6290.19474179985  LUMO (B2g) = -6242.05499998264
> ```

It looks like the output with old versions of libcint. In certain version ofl libcint (4.1 - 4.4.2), a bug in solving linear equation causes the error. However, qcint employs polynomial fittings for the relevant integrals so the error might be ~1e-9 but should not be as large as your test. Can you verify again the path and environment settings of your calculation?

**Comment by sunqm** [comment]
Thanks @susilehtola @bogdanoff  for the detailed runtime environment. Error reproduced. Bug is fixed in qcint v4.4.5

**Comment by susilehtola** [comment]
Yes, now I get the same output for qcint as well
```
init E= -232.398192094587
HOMO (B1g) = -0.251465581587135  LUMO (Au) = 0.0342446621646966
cycle= 1 E= -230.714606240808  delta_E= 1.68  |g|= 0.402  |ddm|= 2.61
HOMO (B1g) = -0.35581581265675  LUMO (Au) = 0.102439491293234
cycle= 2 E= -230.773257120757  delta_E= -0.0587  |g|= 0.106  |ddm|= 0.42
HOMO (B1g) = -0.329722878863841  LUMO (Ag) = 0.109946711454335
cycle= 3 E= -230.777656226318  delta_E= -0.0044  |g|= 0.0267  |ddm|= 0.165
HOMO (B1g) = -0.338380098133097  LUMO (Ag) = 0.108836511682186
cycle= 4 E= -230.777865105423  delta_E= -0.000209  |g|= 0.00638  |ddm|= 0.0223
HOMO (B1g) = -0.337138770908494  LUMO (Ag) = 0.108880229573052
cycle= 5 E= -230.777878224334  delta_E= -1.31e-05  |g|= 0.00101  |ddm|= 0.00604
HOMO (B1g) = -0.336986073316625  LUMO (Ag) = 0.108868838597238
cycle= 6 E= -230.777879032948  delta_E= -8.09e-07  |g|= 0.000259  |ddm|= 0.00197
HOMO (B1g) = -0.336978193903298  LUMO (Ag) = 0.108860235905301
cycle= 7 E= -230.777879085415  delta_E= -5.25e-08  |g|= 2.14e-05  |ddm|= 0.000569
HOMO (B1g) = -0.336977664271867  LUMO (Ag) = 0.108859153860879
cycle= 8 E= -230.777879085681  delta_E= -2.66e-10  |g|= 2.48e-06  |ddm|= 5.47e-05
HOMO (B1g) = -0.336977369785647  LUMO (Ag) = 0.108859245616684
Extra cycle  E= -230.777879085684  delta_E= -3.35e-12  |g|= 7.76e-07  |ddm|= 7.24e-06
converged SCF energy = -230.777879085684
```
